src/main/kotlin/com/coder/gateway/services/CoderSettingsState.kt

```diff
@@ -56,6 +58,10 @@ class CoderSettingsState(
     // connections. This is useful when the hostname used to connect to the
     // Coder service does not match the hostname in the TLS certificate.
     var tlsAlternateHostname: String = "",
+    // Whether to add --disable-autostart to the proxy command.  This works
+    // around issues on macOS where it periodically wakes and Gateway
+    // reconnects, keeping the workspace constantly up.
+    var disableAutostart: Boolean = getOS() == OS.MAC,
```

Comment:
nit: this is more like `defaultDisableAutostart`

Reply:
This gets overridden by Gateway with the value the user has in the settings (through some kind of fancy decorator magic) so it is possible it could be something other than the default.

Reply:
Ahh decorators.